keras_core/layers/layer.py

```diff
@@ -626,6 +627,7 @@ def compute_mask(self, inputs, previous_mask):
     @traceback_utils.filter_traceback
     def __call__(self, *args, **kwargs):
         self._check_super_called()
+        self._called = True
```

Comment:
Why not use `self.built`? Should be equivalent

Reply:
If a user calls `build` by themselves, for example, we should not be allowing export. There is an existing test for this exact flow in TF Keras:

https://github.com/keras-team/keras/blob/master/keras/export/export_lib_test.py#L364

It's a niche case, but let me know what you think.